src/domain/schema/keywords/is-date.keyword.ts

```diff
@@ -0,0 +1,37 @@
+import Ajv from 'ajv';
+
+/**
+ * Registers isDate keyword in AJV
+ *
+ * When the value of isDate is set to true, the field will be validated
+ * according to ISO 8601.
+ *
+ * If the date is valid, the target property is created as {@link Date}.
+ * If the date is invalid, a validation error is thrown.
+ *
+ * Null date values are supported – we assume that a null date is a valid
+ * date.
+ *
+ * @param ajv - the AJV instance to which the isDate keyword should be registered
+ */
+export function addIsDate(ajv: Ajv) {
+  ajv.addKeyword({
+    keyword: 'isDate',
+    compile: (schema) => (value, obj) => {
+      // isDate: false – no need to validate
+      if (schema === false) return true;
+      // Nullability support – if a date is null we skip validation
+      if (value == null) return true;
```

Comment:
For the following schema:
```
date: { type: 'string', isDate: true, nullable: false }
```
if date is null `isDate` returns `true` but then evaluate `nullable` and returns false-invalid schema?

Reply:
The `isDate` does not affect the `nullable` check. We just consider that `isDate` returns `true` if it is `null`.

So it should return invalid schema because the field `date` is `null` when the field is declared as `nullable: false`